### PR TITLE
Add (updated) Pixels API

### PR DESCRIPTION
**Pull request by brentru**
Addressable pixels messaging API to be used between Adafruit IO & WipperSnapper firmware.

* Reduces the complexity of https://github.com/adafruit/Wippersnapper_Protobuf/pull/93 to be less Turing-equivalent and adds flexibility for future API features (addressable pixels, effects + animations)

Addresses https://github.com/adafruit/Wippersnapper_Protobuf/issues/90

### CODE REVIEW

**Review by lorennorman — APPROVED** [review]
i think we're still kicking this around, but it looks reasonable so far

**Comment by brentru** [comment]
@lorennorman  Marking this ready for re-review. `pixels.md` contains sequence diagrams for `pixels.proto`.

**Review by lorennorman — APPROVED** [review]
Liking this a lot. I won't be able to really know how well it's working to communicate things until I sit down to build from it, so expect some feedback through this implementation. I will also learn Mermaid myself and perhaps offer some improvements at that time.

File: proto/wippersnapper/pixels/v1/pixels.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+### Update: NeoPixel
+
+```mermaid
+sequenceDiagram
+autonumber
+IO->>Device: PixelsDeleteRequest
+Note over IO, Device: Contains<br> `pixels_type` of PIXELS_TYPE_NEOPIXEL<br>pixels_pin_data according to DB
+IO-->>Device: PixelsCreateRequest
+Note over IO, Device: Contains:<br> `pixels_type` of PIXELS_TYPE_NEOPIXEL<br>`pixels_num` according to form<br>`pixels_ordering` according to form<br> `pixels_brightness` of 0<br>`pixels_pin_neopixel` according to form<br> `pixels_pin_dotstar_data` is unused<br>`pixels_pin_dotstar_clock` is unused
```
Comment (lorennorman) [inline proto/wippersnapper/pixels/v1/pixels.md:41]: Why brightness 0 here?

File: proto/wippersnapper/pixels/v1/pixels.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+### Update: DotStar
+
+```mermaid
+sequenceDiagram
+autonumber
+IO->>Device: PixelsDeleteRequest
+Note over IO, Device: Contains<br> `pixels_type` of PIXELS_TYPE_DOTSTAR<br>pixels_pin_data according to DB
+IO-->>Device: PixelsCreateRequest
+Note over IO, Device: Contains:<br> `pixels_type` of PIXELS_TYPE_DOTSTAR<br>`pixels_num` according to form<br>`pixels_ordering` according to form<br> `pixels_brightness` of 0<br>`pixels_pin_neopixel` according to form<br> `pixels_pin_dotstar_data` is unused<br>`pixels_pin_dotstar_clock` is unused
```
Comment (lorennorman) [inline proto/wippersnapper/pixels/v1/pixels.md:96]: Brightness 0 here as well?

File: proto/wippersnapper/pixels/v1/pixels.md (hunk 1 above)
Comment (brentru) [reply proto/wippersnapper/pixels/v1/pixels.md:41]: Hm, pixels_brightness can be unfilled here.

File: proto/wippersnapper/pixels/v1/pixels.md (hunk 2 above)
Comment (brentru) [reply proto/wippersnapper/pixels/v1/pixels.md]: Nope, `pixels_brightness ` should be unfilled here. Code doesn't check for this.

**Pull request merged**